### PR TITLE
tilt: global debug flag -> per-request loglevel

### PR DESCRIPTION
**Pull request by landism**
1. use a `loglevel int` instead of `debug bool`, so we can have choose between basic stuff for users (e.g., "starting service") or spammy debug stuff (e.g., "if you ever see this, tell Han she was right about that bug") at runtime
2. attach the log level to the request context instead of setting it globally
3. add one sample usage: `logger.Get(ctx).Verbose("building and deploying")`

### CODE REVIEW

File: internal/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	l.write(VerboseLvl, format, a...)
+}
+
+func (l logger) Debug(format string, a ...interface{}) {
+	l.write(DebugLvl, format, a...)
+}
+
+func (l logger) write(level Level, format string, a ...interface{}) {
+	if l.level >= level {
+		fmt.Printf(format+"\n", a...)
```
Comment (nicks) [inline internal/logger/logger.go]: maybe do a separate fmt.Println("") instead of concatenating strings?

File: internal/engine/upper.go [hunk 2]
```diff
@@ -15,6 +16,7 @@ func UpService(ctx context.Context, service model.Service, watch bool, stdout io
 	if watch {
 		for {
 			// TODO(matt) actually wait for a file to change instead of building on a loop
+			logger.Get(ctx).Verbose("building and deploying")
```
Comment (nicks) [inline internal/engine/upper.go]: do we expect this to be user-visible?

File: internal/proto/daemon.proto [hunk 3]
```diff
@@ -65,9 +64,16 @@ message Output {
...
+enum LogLevel {
+    INFO = 0;
+    VERBOSE = 1;
+    DEBUG = 2;
+}
+
 message CreateServiceRequest {
     Service service = 1;
     bool watch = 2;
+    LogLevel log_level = 3;
```
Comment (nicks) [inline internal/proto/daemon.proto]: i think it's important to be specific of who the expected audience of this log level is. How do I decide which level to use? how do i decide whether to send progress information to the log vs. the CreateServiceReply? there should be with written comments documenting the decision

File: internal/engine/upper.go (hunk 2 above)
Comment (landism) [reply internal/engine/upper.go]: yes, if the user runs with loglevel >= verbose

the message should change (like, at least include a service name), but this PR is more about allowing logging than polishing the UI

File: internal/proto/daemon.proto (hunk 3 above)
Comment (landism) [reply internal/proto/daemon.proto]: re: log vs CreateServiceReply - my plan is to hook the logger up to the CreateServiceReply. maybe the name "logger" is misleading if this is what we're using to report output to the user?

I've added comments to the `Logger` interface, which might be the least bad place to put them, but open to thoughts.

File: internal/logger/logger.go (hunk 1 above)
Comment (landism) [reply internal/logger/logger.go]: is there an advantage to that, or a matter of taste?

(fwiw, that option kinda goes away when this is hooked up to the grpc stream)

**Pull request merged**